### PR TITLE
Add simple script to compile heron and package it in a docker image.

### PR DESCRIPTION
**Pull request by cckellogg**
This PR augments build-artifacts.sh to generate docker images for the platform being compiled. The docker image generated contains both the heron core and tools to simplify deployment. This docker image will be published to docker hub and it will be used by DC/OS marathon and Kubernetes.

### CODE REVIEW

File: docker/Dockerfile.dist.ubuntu14.04 [hunk 1]
```diff
@@ -27,6 +29,12 @@ ADD artifacts /heron
 WORKDIR /heron
 
 #run heron installers
-RUN /heron/heron-tools-install-*-ubuntu14.04.sh && \
-    /heron/heron-client-install-*-ubuntu14.04.sh && \
-    /heron/heron-tools-install-*-ubuntu14.04.sh
\ No newline at end of file
+RUN /heron/heron-tools-install-$heronVersion-ubuntu14.04.sh && \
+    /heron/heron-client-install-$heronVersion-ubuntu14.04.sh && \
+    /heron/heron-tools-install-$heronVersion-ubuntu14.04.sh
```
Comment (cckellogg) [inline docker/Dockerfile.dist.ubuntu14.04]: removing

File: docker/build-heron-docker.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (cckellogg) [inline docker/build-heron-docker.sh]: how about build-heron-docker-image.sh?

File: docker/Readme.md [hunk 3]
```diff
@@ -12,6 +12,12 @@
 # e.g. ./docker/build-docker.sh ubuntu14.04 testbuild ~/heron-release
 ```
 
+# To compile source and build a docker image for running heron daemons:
+```
+./docker/build-docker.sh <platform> <version_string> <output-directory>
+# e.g. ./docker/build-heron-docker.sh ubuntu14.04 testbuild ~/heron-release
```
Comment (cckellogg) [inline docker/Readme.md]: fixing

File: docker/build-artifacts.sh [hunk 4]
```diff
@@ -90,7 +90,11 @@ verify_source_exists() {
 }
 
 setup_scratch_dir() {
-  mkdir -p $1
+  if [ ! -f "$1" ]; then
```
Comment (kramasamy) [inline docker/build-artifacts.sh]: No check is required.

File: docker/build-artifacts.sh [hunk 5]
```diff
@@ -90,7 +90,11 @@ verify_source_exists() {
...
+    mkdir $1
+    mkdir $1/artifacts
```
Comment (kramasamy) [inline docker/build-artifacts.sh]: use mkdir -p

File: docker/build-artifacts.sh [hunk 6]
```diff
@@ -99,6 +99,23 @@ setup_output_dir() {
   mkdir -p $1
 }
 
+run_build_docker() {
```
Comment (kramasamy) [inline docker/build-artifacts.sh]: run_build_docker please change build_docker_image

**Pull request merged**